### DISCUSSION TITLE
Is there a way to check if form was touched in real time?

### DISCUSSION THREAD

**Discussion by piotrkulpinski** [airjp73/rvf#236] [Q&A]
Hi, 

I want to check if any of the fields in the form was changed in real time (input's onChange).
Currently, I'm using something like this, but it only changes the touchedFields onBlur.

```ts
const hasChanged = () => {
  const { touchedFields, } = useFormContext()
  return Object.keys(touchedFields).length > 0
}
```

Is there any way to tell the form to check this onChange instead of onBlur?

**Comment by airjp73** [comment]
Hi!

What you're talking about sounds bit more like a `dirty` state rather than a `touched` state. There's an issue on that [here](https://github.com/airjp73/remix-validated-form/issues/104) with a workaround in the comments.
